### DISCUSSION TITLE
fix for being stuck on "AI Ready" screen

### DISCUSSION THREAD

**Discussion by J-Massari** [jayofelony/pwnagotchi#156] [Q&A] [answered]
i'm unsure how to proceed due to info on the fix being from 3 years ago on reddit. any advice would be greatly appreciated.

**Comment by jayofelony** [comment]
Lets start with the image version, what device and any error logging.

**Reply by J-Massari** [reply]
Hardware: R PI Z W/H, wave share V4, SanDisk 32GB Micro-SD.

SOFTWARE: Jayofelony PWNagotchi build v2.8.9 32bit

ISSUES: device sat on "AI MODE READY" for 4 hours, after some updating now sits on "NUERAL NETWORK READY". Upon looking into some of the CLI it seems my WLAN0 may not be working.

just curious if these lines in the config could be an issue (#dtoverlay=disable-wifi):

```
[all]
dtoverlay=dwc2
dtparam=i2c1=on
dtparam=i2c_arm=on
dtparam=spi=on
gpu_mem=1
dtoverlay=dwc2
#dtoverlay=disable-wifi

[pi0]
dtoverlay=spi1-3cs
#dtoverlay=disable-wifi
```

I also can not discover wlan0 what so ever.

`iwconfig` only shows:

```
lo no wireless extensions.

usb0 no wireless extensions.
```
`sudo rfkill unblock wifi `has no returns.

`iwlist wlan0 scan` returns:

`interface does not support scanning.`
`dmesg | grep wifi` has no returns.

`cat /sys/firmware/devicetree/base/model` returns:

`Raspberry Pi Zero W Rev 1.1`
`ifconfig -a` returns:

```
lo: flags=73<UP,LOOPBACK,RUNNING> mtu 65536
inet [127.0.0.1](http://127.0.0.1/) netmask [255.0.0.0](http://255.0.0.0/)
inet6 ::1 prefixlen 128 scopeid 0x10<host>
loop txqueuelen 1000 (Local Loopback)
RX packets 1854 bytes 92700 (90.5 KiB)
RX errors 0 dropped 0 overruns 0 frame 0
TX packets 1854 bytes 92700 (90.5 KiB)
TX errors 0 dropped 0 overruns 0 carrier 0 collisions 0

usb0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST> mtu 1500
inet [10.0.0.2](http://10.0.0.2/) netmask [255.255.255.0](http://255.255.255.0/) broadcast [10.0.0.255](http://10.0.0.255/)
inet6 fe80::c7e2:bc8a:9d13:86d6 prefixlen 64 scopeid 0x20<link>
ether 6e:62:7a:0a:9e:f7 txqueuelen 1000 (Ethernet)
RX packets 1268 bytes 92122 (89.9 KiB)
RX errors 0 dropped 0 overruns 0 frame 0
TX packets 1228 bytes 171202 (167.1 KiB)
TX errors 0 dropped 0 overruns 0 carrier 0 collisions 0
```

**Reply by renewedbullet** [reply, marked as answer]
the dtoverlay=disable-wifi is for disabling onboard wifi and when using external wifi card. I would fresh flash with RPI imager, make sure you dont extract the zip file the image comes in just use it as is. and as soon as it boots run IWconfig, if you have no interfaces something is wrong with the pi. You could also load up a generic pi image from RPI imager and see if iwconfig works and display a interface.
